### PR TITLE
Implements sha3Raw and soliditySha3Raw

### LINKED ISSUE #1961: web3.utils.sha3 and web3.utils.soliditySha3 return null when an empty byte sequence is hashed
Based on the code, this appears to be done intentionally for some reason, but it seems like a very bad idea to return the wrong answer. There's also no mention of this non-standard behavior in the documentation.

From https://github.com/ethereum/web3.js/blob/1.0/packages/web3-utils/src/utils.js#L434,L435:

```
var SHA3_NULL_S = '0xc5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470';
...
    if(returnValue === SHA3_NULL_S) {
        return null;
```

Removing those lines would fix the problem.

### PR DESCRIPTION
**Pull request by nivida**
## Description

Fixes #1961

This PR does fix the referenced "bug" above where you get returned ``null`` instead of the hash.

@cgewecke I'm open for better function names ;)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] Any dependent changes have been merged and published in downstream modules.
- [x] I ran ```npm run test``` with success and extended the tests if necessary.
- [x] I ran ```npm run build``` and tested the resulting file from ```dist``` folder in a browser.
- [ ] I have tested my code on the live network.

### CODE REVIEW

File: packages/web3-utils/src/soliditySha3.js [hunk 1]
```diff
@@ -241,5 +241,18 @@ var soliditySha3 = function () {
     return utils.sha3('0x'+ hexArgs.join(''));
 };
 
+/**
+ * Hashes solidity values to a sha3 hash using keccak 256 but does return the hash of value `null` instead of `null`
+ *
+ * @method soliditySha3
```
Comment (cgewecke) [inline packages/web3-utils/src/soliditySha3.js]: Just a tiny thing here...
```suggestion
 * @method soliditySha3Raw
```

**Review by cgewecke — APPROVED** [review]
This LGTM! Really nice. 

Only thing I can think of that might be useful to add to the tests is maybe one non-empty-string test case borrowed from the other sha tests, just so the basic functionality is covered in case someone accidentally changes logic later.
